### DISCUSSION TITLE
Ping/Pong and idleTimeout

### DISCUSSION THREAD

**Discussion by uasan** [uNetworking/uWebSockets.js#195] [Q&A] [answered]
Hi.
By [RFC6455](https://tools.ietf.org/html/rfc6455#section-5.5.2)  

> The Ping frame contains an opcode of 0x9.
> An endpoint MAY send a Ping frame any time after the connection is
> established and before the connection is closed.
> ...
> Upon receipt of a Ping frame, an endpoint MUST send a Pong frame in
> response, unless it already received a Close frame.
> ...
> A Ping frame may serve either as a keepalive or as a means to
> verify that the remote endpoint is still responsive.

Current behaviour in uWebSockets, after received `ping` frame, none response `pong`, and close connection on expired `idleTimeout`, this by design or bug?

**Comment by ghost** [comment, marked as answer]
This sounds wrong, responding to ping is something you have to do to pass Autobahn
